src/routes/resources/+page.server.ts

```diff
@@ -1,20 +1,31 @@
-import type { PageServerLoad } from './$types'
-import { parse } from 'yaml'
-import fs from 'fs'
+import type { PageServerLoad } from "./$types";
+import { parse } from "yaml";
+import fs from "fs";
 
-let resources_path: string = "data/resources.yml"
+let resources_path: string = "data/resources.yml";
 
 // Reading in resources
-const file = fs.readFileSync(resources_path, 'utf8')
-const yml_data = parse(file) // TODO: Create interface for this, and then use it to validate edits during CI/CD
+const file = fs.readFileSync(resources_path, "utf8");
+const yml_data = parse(file); // TODO: Create interface for this, and then use it to validate edits during CI/CD
 
 // Creating a list of unique tags
 let tags: string[] = [];
 let resource;
 for (resource of yml_data) {
-  tags.push(...resource.tags)
+  tags.push(...resource.tags);
+}
+tags = [...new Set(tags)];
+
+// create list of unique tags with count
+let tag_count: { [key: string]: number } = {};
+for (const tag of tags) {
+  tag_count[tag] = 0;
+}
+for (const resource of yml_data) {
+  for (const tag of resource.tags) {
```

Comment:
Love this simple histogram type solution. I was messing around with using sets but this is clearly the simpler solution.

This makes me wonder if we should limit the number of filter tags that we read in from the resource file given the opportunity for 0(n^2) complexity here. Assuming resource list could be come very large in the future. We might want to limit the number of tags read to like 10 so we can assume O(10N) for the resources. This also seems reasonable for the sets matching logic when we go to filter.